HACKING.md

```diff
@@ -79,96 +79,197 @@ The bigger PR will likely take some time to get reviewed, and discussed, while s
-  parameters are `under_scored`.
-  * If you introduce a new namespace, you can use underscore as
-    namespace-identifier separator, for example
-    `StrList_BinarySearch()`.
-  * Avoid introducing extra long identifiers, like
-    ~~`GenericAgentConfigParseWarningOptions()`~~.
-* Try not to include assignments inside if/while expressions
-  *unless they avoid great repetition*. On the average case,
-  just put the assignment on the previous line. So try NOT to do
-  the following:
-  ~~`if ((ret = open(...)) == -1)`~~
-* Explicit comparisons are better than implicit, i.e. prefer writing
-  `if (number == 0)` or `if (pointer == NULL)`
-  instead of ~~`if (!number)`~~ or ~~`if (!pointer)`~~. It only makes
-  sense to test booleans directly, for example `if (is_valid)` is good.
-  Furthermore have the literal last in the comparison, not first, i.e.
-  prefer writing `if (open(...) == -1)` instead of
-  ~~`if (-1 == open(...))`~~.
-* Control statements need to have braces on separate line,
-  no matter how simple they are.
-  * Caution, do-while loops should have the closing brace at the same
-    line with while, so that it can't be confused with empty while statement.
-    ```c
-    do
-    {
-        // ...
+Our coding style is loosely based on Allman-4 and the [Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html).
+Keep in mind that these are guidelines, there will always be some situations where exceptions are appropriate.
 
-    } while (condition);
-    ```
-* In functions which can fail, error code (int) or success/failure (bool) should be returned.
-    * `true`(bool) and `0`(int) should always signify success.
-    * Only return an error code (int) when there are multiple different return values for different errors. If a function can only return `0` (success) or `-1` (error) use `bool` instead.
-    * Compiler can enforce checking of return value, output of function can be in an output parameter (pointer).
-* *C99 is encouraged in the language, use it.*
-
-  As for using C99-specific libc functions, you can mostly use them,
-  because we provide replacement functions in libcompat, since many old
-  Unix platforms are missing those. If there is no replacement for a
-  C99-specific function, then either stick to C89, or write the
-  libcompat replacement.
-
-  Current functions known to be missing from libcompat
-  (so stick to C89):
-  * `[s]scanf()`
-* Fold new code at 78 columns.
-* Do not break string literals. Prefer having strings on a single line
-  In order to improve grep-ability. If they do not fit on a single line,
-  try breaking on punctuation. In worst case scenario, you are allowed
-  to surpass the 78 columns limit.
+### Formatting / Whitespace
 
-  Bad:
+* Feel free to use `clang-format` to format new code you submit.
+  (Our configuration is in `.clang-format`).
+* 4 spaces per indentation, no tabs.
+* Fold new code at 78 columns.
```

Comment:
I still hate this.

Reply:
Let's do a separate discussion or PR for changing it then :)